### DISCUSSION TITLE
How to add database capabilities to a lightweight-service app?

### DISCUSSION THREAD

**Discussion by NickUfer** [loco-rs/loco#815] [Q&A]
`loco new` has 3 presets, `lightweight-service`, `Rest API` and `SaaS app`. For me personally the Rest API preset has too many unnecessary things enabled by default. So my idea was to use the lightweight-service and enable a bare bones rest api and a database.
My problem is that I can't seem to find any documentation on how to do this. Also `cargo loco generate` seems to remove generation options based on some to me unknown criteria. Is there some documentation about how to deal with this? Enabling new features after the first project generation?

**Comment by jondot** [comment]
You can safely use the rest-api, just FYI all projects are lightweight. If you take a look at the dependencies, you'll discover that everything is the bare essentials.
Add to this the fact that we're using Rust and you shouldn't have any concept of "heaviness".

The only thing to think about is the *mental* model. When you don't need a DB you don't need the `models` folder and other folders, and this is why we've created the lightweight service. 

Otherwise in terms of code, binary, if you go for the Rest API template you'll get the same thing as if you've added a DB to the lightweight service.
